### PR TITLE
Prepare 0.29.0 release

### PR DESCRIPTION
**Pull request by Misty-W**
## Description

---

### License

- [x] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.
- [x] I added unit tests for new code.
- [x] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [x] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [x] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.
- [x] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

**Comment by Misty-W** [comment]
Btw,  after creating this PR I merged a Stim upgrade and am waiting for the auto-rebase of the Stimcirq upgrade to pass before merging that as well. Just holding the update to the dependency list until then.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,67 @@
 # Changelog
 
+## Version 0.29.0
+
+### Summary
+
+#### Update Pauli Twirling
+Thanks to @purva-thakre for updating Mitiq's PT functions, clarifying that PT is a noise tailoring technique and for consolidating utilities to be shared between PT and other techniques.
```
Comment (natestemen) [inline CHANGELOG.md:8]:
```suggestion
Thanks to @purva-thakre for updating Mitiq's PT functions, clarifying that PT is a noise tailoring technique and for consolidating utilities to be shared between PT and other techniques.
This release removes the `execute_with_pt` functionality in replace of `pauli_twirl_circuit`.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,67 @@
...
+```py
+from mitiq.pt.pt import pauli_twirl_circuit
+
+pauli_twirl_circuit(circuit)
+```
+
+#### Classical Shadows
+
+Top-level functions and tests for classical shadows estimation are now available in Mitiq.
+Congrats @min-li on completing the main functionality for this technique!
```
Comment (natestemen) [inline CHANGELOG.md:19]: Lets add a short note that documentation on this method is not yet out, but coming soon!

**Review by natestemen — APPROVED** [review]
Nice work with the release notes! I really like the little demos you give, and the compliments to the many people who helped with this release. I think it's really nice.

I have two small recommendations, but please do as you see fit 🙂

File: CHANGELOG.md (hunk 1 above)
Comment (Misty-W) [reply CHANGELOG.md:8]: Agree, good to be precise about what changed. Added in latest commit.

File: CHANGELOG.md (hunk 2 above)
Comment (Misty-W) [reply CHANGELOG.md:20]: Good idea, added in latest commit.

**Comment by Misty-W** [comment]
thanks @natestemen! PR is updated accordingly, just waiting for the checks to pass 😄

**Pull request merged**